src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -288,27 +442,86 @@ object JsonProtocol {
           throw new InvalidAnnotationJSONException(s"Expected field 'class' not found! $obj")
         case JObject(fields) => findTypeHints(fields.map(_._2))
         case JArray(arr)     => findTypeHints(arr)
-        case oJValue         => Seq()
+        case _               => Seq()
       })
       .distinct
 
+    // I don't much like this var here, but it has made it much simpler
+    // to maintain backward compatibility with the exception test structure
+    var classNotFoundBuildingLoaded = false
     val classes = findTypeHints(annos, true)
-    val loaded = classes.map(Class.forName(_))
+    val loaded = classes.map { x =>
+      try {
+        Class.forName(x)
+      } catch {
+        case _: java.lang.ClassNotFoundException =>
+          classNotFoundBuildingLoaded = true // tells us which Exception to throw in recovery
+          // Found an annotation we don't recognize, So add UnrecognizedAnnotation to `loaded`
+          Class.forName("firrtl.annotations.UnrecognizedAnnotation")
+      }
+    }
     implicit val formats = jsonFormat(loaded)
-    read[List[Annotation]](in)
-  }).recoverWith {
+    try {
+      read[List[Annotation]](in)
+    } catch {
+      case e: org.json4s.MappingException =>
+        // If we get here, the build `read` failed to process an annotation
+        // So we will map the annos one a time, wrapping the JSON of the unrecognized annotations
+        var exceptionList = new mutable.ArrayBuffer[String]()
```

Comment:
```suggestion
        val exceptionList = new mutable.ArrayBuffer[String]()
```
It's already mutable, it should be a val.

Reply:
yup